### PR TITLE
#WYfBlp40 Setup continuous integration and Heroku Pipeline

### PR DESCRIPTION
**Pull request by Niyiragira**
#### What does this PR do?
Requests to merge Ch-frontend-setup-continuous-integration-and-testing-WYfBlp40 into develop

#### Description of Task to be completed?

- Setup Heroku pipeline

- Configured Testing environment with Jest & Enzyme

- Travis CI is configured as a continuous integration service

- Travis CI badge has been added to your repo

- Configure CodeClimate for code analysis and Maintainability

#### How should this be manually tested?
-clone the repo and checkout origin/Ch-frontend-setup-continuous-integration-and-testing-WYfBlp40 

- run `npm install` 

- run `npm test`

- Go to [Heroku](https://cabal-fronte-ch-fronten-w7str4.herokuapp.com/) to see deployment

- check the travis and codeclimate badges  on This [Branch](https://github.com/atlp-rwanda/cabal-phantom-frontend/tree/Ch-frontend-setup-continuous-integration-and-testing-WYfBlp40)

#### What are the relevant pivotal tracker stories?
[WYfBlp40](https://trello.com/c/WYfBlp40/27-setup-continuous-integration-quality-assurance-pipeline)

#### Screenshots (if appropriate)
Screen Shot 2020-11-24 at 10 57 49

### CODE REVIEW

**Review by Nkbtemmy — DISMISSED** [review]

**Review by ncutixavier — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "url": "git+https://github.com/atlp-rwanda/cabal-phantom-frontend.git"
+  },
+  "author": "cabal Team",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/atlp-rwanda/cabal-phantom-frontend/issues"
+  },
+  "homepage": "https://github.com/atlp-rwanda/cabal-phantom-frontend#readme",
+  "dependencies": {
+    "@webpack-cli/init": "^1.0.3",
```
Comment (ceelogre) [inline package.json]: Why is this in dependencies?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  "devDependencies": {
+    "babel-core": "^6.26.3",
+    "babel-loader": "^8.2.0",
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-react": "^6.24.1",
+    "html-webpack-plugin": "^4.5.0",
+    "jest": "^26.6.3",
+    "react-hot-loader": "^4.13.0",
+    "react-scripts": "^4.0.0",
+    "typescript": "^2.9.2",
```
Comment (ceelogre) [inline package.json]: What are you using typescript for?

File: src/test/app.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import App from '../components/app'
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+describe("Tests if all components are being rendered and work as expected", () => {
+    it("Whole app component should render correctly", () => { 
+        const div = document.createElement('div');
+        ReactDOM.render(<App />, div);
```
Comment (ceelogre) [inline src/test/app.test.js]: There's no assertion here. You're not testing for anything.

**Review by ceelogre — CHANGES_REQUESTED** [review]
Reach out to Emmanuel so you can solve the issues together. Your work is ready.

File: src/test/app.test.js (hunk 3 above)
Comment (Niyiragira) [reply src/test/app.test.js]: You can check again 
Corrected
Now I am testing for something.

**Review by ncutixavier — DISMISSED** [review]

File: webpack.config.js [hunk 4]
```diff
@@ -38,7 +38,7 @@ module.exports = {
       },
       {
         test: /\.scss$/,
-        use: ["style-loader", "css-loader", "sass-loader"]
+        use: ["style-loader", "css-loader"]
```
Comment (Nkbtemmy) [inline webpack.config.js]: why don't you add sass-loader???

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 4 above)
Comment (Niyiragira) [reply webpack.config.js]: It is there now

**Review by Nkbtemmy — DISMISSED** [review]

**Review by karigirwa — DISMISSED** [review]

**Review by ncutixavier — DISMISSED** [review]

**Review by ceelogre — APPROVED** [review]

**Pull request merged**